lib/vnet/app_resolver.go

```diff
@@ -159,21 +166,39 @@ func (r *TCPAppResolver) resolveTCPHandlerForCluster(
 		Limit:               1,
 	})
 	if err != nil {
-		// Don't return an error so we can try to find the app in different clusters or forward the request
-		// upstream.
+		// Don't return an unexpected error so we can try to find the app in different clusters or forward the
+		// request upstream.
 		slog.InfoContext(ctx, "Failed to list application servers.", "error", err)
-		return nil, false, nil
+		return nil, ErrNoTCPHandler
```

Comment:
Did you see the issue with no relogin prompt when connecting to an app after the cert has expired? https://github.com/gravitational/teleport/pull/41031#discussion_r1609684247

I don't know if this is something we'd like to address at this point or somewhere else.

Reply:
sorry i thought i replied to that one last week, must have forgotten to hit send

Will need to figure out exactly where to add a RetryWithRelogin to handle expired profiles but not necessarily profiles that were never logged into in the first place. It will probably require an addition to the AppProvider interface

this PR is already fairly large and would be better to get in sooner than later, I'd like to handle this in another PR if that's alright